src/openforms/formio/rendering/nodes.py

```diff
@@ -143,7 +143,9 @@ def is_visible(self) -> bool:
                 self.component, artificial_repeating_group_data
             ):
                 return False
-        elif not is_visible_in_frontend(self.component, self.step_data):
+        elif self.mode != RenderModes.registration and not is_visible_in_frontend(
```

Comment:
you can use

```suggestion
        elif self.mode is not RenderModes.registration and not is_visible_in_frontend(
```

(I see this `!=` / `==` is used as well in the function already, I think they can be changed as well)

Reply:
I don't fully agree with this, it's down to an implementation detail of how enum is used, but I think in practical reasons you can compare this is a string comparison which is typically done with `==` and `!=`. Let's stick to the existing usage rather than changing the existing code.

Reply:
Well according to docs equality comparison seems to be the implementation detail, not identity (`Enumeration members are compared by identity` / `Equality comparisons are defined though`).

But fine to keep it as is

Reply:
> Equality comparisons are defined though:

That's explicit support for equality comparisons, which is different than an implementation detail. It means you can rely on it and it's guaranteed to work as expected, or will come with appropriate release notes if the behaviour changes :wink: 

The fact that enum is used under the hood however is the implementation detail, as that happens to be how Django implements its `models.*Choices` classes. Unfortunately Django docs don't seem to give a comparison example, they do give examples of comparing `member in {Foo.bar, Foo.baz}` though. Which in turn leads to the implementation detail of how `set.__contains__` is implemented: identity or equality. I suspect it's the latter.

Reply:
> The fact that enum is used under the hood however is the implementation detail

Ah ok I thought you meant the identity comparison is the implementation detail of the enum class, but when taking django choices into account it makes sense